tests/TestCase.php

```diff
@@ -139,6 +140,11 @@ public function provideInvalidIntegerValues()
         return $this->wrapValuesForDataProvider($this->getInvalidIntegerValues());
     }
 
+    public function provideInvalidUpdateValues()
```

Comment:
Is this only used for Operation classes? If so, you can consider moving it to `MongoDB\Tests\Operation\TestCase`, which is where I put other providers.

Reply:
Ah, good point. I've moved this and `getInvalidUpdateValues` for consistency.

This also provides a nice refactoring opportunity down the line once we make all our data providers static (which is required in later PHPUnit versions): we can offload these data providers to separate classes or traits and import them as necessary.

Reply:
Yes, static data providers have been allowed for quite some time but have only become required as of PHPUnit 10. We could make that change already.